### ISSUE TITLE
React SPFx ADAL - Unable to get a token when multiple webparts are placed on a page.

### ISSUE THREAD

**Issue by sohotra** [SharePoint/sp-dev-fx-webparts#361]
#### Category
- [x] Question
- [ ] Bug
- [ ] Enhancement

I'd like to run multiple webparts on the same page that call out to Graph for their respective token. The problem that is occurring is on first load (no cache), only one of the webparts does the login and goes into the callback, the other webpart just doesn't do anything. It's sort of a race condition, whichever webpart gets to it first gets the token and the other one doesn't do anything.

One way to recreate this behaviour is take any webpart that calls out to graph for a token, in the tenant workbench (<tenant>.sharepoint.com/_layouts/15/workbench.aspx) add the same webpart twice.

I should also add that there is no Sign-In button, we need the webparts to just login and load the data.

**Comment by waldekmastykarz** [comment]
Have you read this doc page which discusses the issue of using ADAL JS with web parts and provides a workaround for it? https://docs.microsoft.com/en-us/sharepoint/dev/spfx/web-parts/guidance/call-microsoft-graph-from-your-web-part

**Comment by sohotra** [comment]
Yes, I used this as my main reference to write out the code. I can verify that in my local storage I saw a guid underscore and then the adal keys, so I was able to build it such that AuthenticationContext is not acting as a singleton. I believe the issue is when the login happens, all the webparts try to log in.. but only one gets to do it. If I add the web parts one after another, there is some delay so they log in and get a token. Clear local storage, refresh and it will not work.

Is this something that is supposed to work and I just coded it wrong or this is an issue ?
